great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -290,9 +290,7 @@ def run(
                 reconciliation_directives=reconciliation_directives,
             )
             self.rule_states.append(rule_state)
-
-            domains_count += len(rule_state.domains)
-            progress_bar.update(domains_count)
+            progress_bar.update(1)
```

Comment:
@Shinnnyshinshin I see the benefit of the fix, but I was hoping we could update the progress bar on every `Domain`, because that is the lowest quantum of progress.  I am approving this for now so we can merge it, but if my request is possible in reasonable time effort, that would be awesome (it might require some kind of coordination  between `RuleBasedProfiler.run()` and `Rule.run()`.  Thanks!

Reply:
I did look this, but it does require one more 'layer' of computation to do the sum calculation ( rules --> rule --> domain ), which I was trying to avoid. Would keeping it at the Rule layer still be ok?